compiler/src/iree/compiler/Codegen/Transforms/Transforms.cpp

```diff
@@ -627,13 +633,142 @@ struct FoldReshapeIntoInterfaceTensorStore
     return success();
   }
 };
+
+/// Folds tensor.collapse_shape with static shape into the source
```

Comment:
Why do the existing patterns not already handle this?

Reply:
The existing patterns only apply to full slices, i.e. `flow.tensor.dispatch.store tensor<1x2x3> into tensor<1x2x3>`. This is trying to handle cases where we are inserting a partial slice.

Reply:
Could you just modify that to handle offsets. Looking at this you are only looking at contiguous slices, with the full slice essentially being offsets being all zero... (You can also just replace the previous one with this).

Reply:
Done. The previous pattern still remains because it was also handling expand_shape.